### ISSUE TITLE
Support different validations for the same segment at different levels

### ISSUE THREAD

**Issue by benisantos** [xlate/staedi#78]
I'm not sure if this is a new feature or a question... Our problem is:
- Some segments (with the same segment name) have different validation rules depending on the level they are.

One example from PADIS EDIFACT (spec [here](https://www.iata.org/contentassets/18a5fdb2dc144d619a8c10dc1472ae80/pnrgov20edifact20implementation20guide2016_1.pdf)):

- Segment "TVL" at Level 0 (page 57): many mandatory fields
- Segment "TVL" in Gr5 at Level 2 and Gr.12 at Level 4 (page 59): same fields, but less number of mandatory ones

Does the transaction schema support different validation rules depending on the level/loop of the segment?

**Comment by MikeEdgar** [comment]
@benisantos - this is supported using "implementation" schemas. Take a look at the Validation page on the [wiki](https://github.com/xlate/staedi/wiki). Basically, you define the `transaction` schema as the base standard, then you customize the segments using `<implementation>`. If it ends up being too large, you can always defined the implementation in a separate file and include the other schema that defines the `transaction` and all the segment and element types.

That example in the wiki is very basic (just a single segment), but the idea applies to both segments and loops. Let me know if you have further questions. The example could probably be a bit more complex.

**Comment by benisantos** [comment]
I'm taking a look at `src/test/resources`, but I have a doubt:
- There are examples of reusing a transaction segment in the implementation without changes. For instance:
```xml
  <segment type="AK2">
    <!-- Unchanged from standard -->
  </segment>
```
- But I don't find an example of reuse a full **loop** in the same way. Is supported something like that on an implementation?
```xml
  <loop code="foo" type="bar">
    <!-- Unchanged from standard -->
  </loop>
```

**Comment by MikeEdgar** [comment]
@benisantos - that is not supported. Typically, implementations of a loop will reduce the occurrences of segments or remove them altogether, so I don't think there would really be a case for using the loop as-is. In the case of PNRGOV, the document you linked is the implementation guide. I have not been able to locate the standard definition, but I believe it would likely be a more general description of the message. For example, all references to `TVL` would be identical. It's only in the implementation guide where the different `TVL` requirements are specified.

Do you know if the PNRGOV standard is available anywhere?

**Comment by benisantos** [comment]
@MikeEdgar I don't know any standard documentation like you suggest :(
I'll try your transaction-vs-implementation advice as soon as I can.
Anyway, Thank you for your help!

**Comment by benisantos** [comment]
@MikeEdgar I think I found an error related to transaction-vs-implementation schemas and empty segments (like `SRC` in PNRGOV), when the empty segment is the first element of the loop (in other positions works perfectly).

If I change the transaction and implementation schemas to remove `SRC` and remove as well from the EDIFACT file, the validation is OK. Same, if I put `SRC` as second or third position of the loop. But in first position, the validation process fails.

Could you realize where the problem is? Or do you need a failed test to help you?

**Comment by MikeEdgar** [comment]
@benisantos , I will take a look later today, but a test case would definitely be appreciated!

**Comment by benisantos** [comment]
@MikeEdgar I have created a custom issue (#79) for the problem with empty segments starting loops. And I have attached a failed test case :)

**Comment by MikeEdgar** [comment]
@benisantos - how are things going with your implementation using the PNRGOV guide? I wanted to check if there is anything still needed on this issue.

**Comment by benisantos** [comment]
Sorry @MikeEdgar , I'm busy with other business now :( I'll go back to this issue as soon as I can.

**Comment by MikeEdgar** [comment]
Hi @benisantos - I just wanted to follow up on this one to see if you were able to solve your schema issue.

**Comment by benisantos** [comment]
Hi @MikeEdgar ! Yes, the schema issue was solved with the advice of question #104. This question could be closed :)

And thank you for this awesome lib!

**Comment by MikeEdgar** [comment]
@benisantos , thank you, glad to hear that!
